### PR TITLE
Releases and binaries build automation

### LINKED ISSUE #431: New releases and binaries build automation
At the moment we write a RR version on the linking stage with `LDFLAGS` manually. And forming releases also manually. Now, it's time to move forward. The scope of this issue is the following:
1. Support automatic version creation on the CI stage.
2. Automated releases and binaries build.

### PR DESCRIPTION
**Pull request by tarampampam**
This PR contains:

- [x] CI steps for automatic release distributive creation and uploading to github release
- [x] CI step for automatic docker image creation (on GitHub side, not `hub.docker.com`)
- [x] `build.sh` script removal
- [x] Binary file building in `Dockerfile` using `go build` (not `build.sh`)
- [x] Docker labels with app version and build time
- [x] `Makefile` cleaning
- [x] `./bin/rr` now do not depends from `build.sh` file and use [`composer/package-versions-deprecated`](https://github.com/composer/package-versions-deprecated) package for "self version reading"

> :warning: Steps, that must be done before merging described [here](https://github.com/spiral/roadrunner/pull/433#issuecomment-740426206)

Closes #431

### CODE REVIEW

**Comment by tarampampam** [comment]
I'll tested this CI action on my fork, and all works fine. Opened question is "should we remove `build.sh` script as useless for us now?"

**Review by 48d90782 — APPROVED** [review]
LGTM

**Comment by 48d90782** [comment]
> I'll tested this CI action on my fork, and all works fine. Opened question is "should we remove `build.sh` script as useless for us now?"

Yep, we should remove this redundant script if all work by providing binaries will be handled by CI.

**Comment by tarampampam** [comment]
`build.sh` removed. PR ready for review now!

What we need to do before merging:

- [ ] Create repository secret `DOCKER_LOGIN` with [hub.docker.com](https://hub.docker.com/) username
- [ ] Create repository secret `DOCKER_PASSWORD` with token (**not** regular password) for [hub.docker.com](https://hub.docker.com/) (can be taken on [this page](https://hub.docker.com/settings/security))

Automatic disrtibutive creation was tested on my fork (file names, format and content are same as it was):

image

File: Dockerfile [hunk 1]
```diff
@@ -1,27 +1,46 @@
-FROM golang:1.15.5 as builder
+# Image page: <https://hub.docker.com/_/golang>
+FROM golang:1.15.5-alpine as builder
```
Comment (48d90782) [inline Dockerfile:2]: I would prefer using just 1.15.5 without alpine. It uses Debian by default and a little big bigger, but from my POV should be more predictable.

File: Dockerfile [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
+# download dependencies and compile binary file
 RUN set -x \
-    && apt-get update -y \
-    && apt-get install -y bash git \
-    && go version \
-    && bash ./build.sh \
-    && test -f ./.rr.yaml
+    && go mod download \
+    && go mod verify \
+    && CGO_ENABLED=0 go build -trimpath -ldflags "$LDFLAGS" -o ./rr ./cmd/rr/main.go
```
Comment (48d90782) [inline Dockerfile:22]: To use the advantage of docker caching layers, it's better to separate this command into 2:
RUN go mod download + go mod verify
RUN build

**Review by 48d90782 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (tarampampam) [reply Dockerfile]: yep, done

File: Dockerfile (hunk 2 above)
Comment (tarampampam) [reply Dockerfile]: done

**Review by 48d90782 — APPROVED** [review]

**Pull request merged**